File: [CODE]/rightoversJAVA/src/main/java/sep3/g3/rightoversjava/model/dto/OpeningHoursCreationDTO.java
package sep3.g3.rightoversjava.model.dto;

import java.time.LocalTime;


public class OpeningHoursCreationDTO {

    public String username;
    private final LocalTime mondayOpeningHours;
    private final LocalTime mondayClosingHours;
    private final LocalTime tuesdayOpeningHours;
    private final LocalTime tuesdayClosingHours;
    private final LocalTime wednesdayOpeningHours;
    private final LocalTime wednesdayClosingHours;
    private final LocalTime thursdayOpeningHours;
    private final LocalTime thursdayClosingHours;
    private final LocalTime fridayOpeningHours;
    private final LocalTime fridayClosingHours;
    private final LocalTime saturdayOpeningHours;
    private final LocalTime saturdayClosingHours;
    private final LocalTime sundayOpeningHours;
    private final LocalTime sundayClosingHours;

    private  boolean mondayIsOpen;

    private boolean tuesdayIsOpen;

    private boolean wednesdayIsOpen;

    private boolean thursdayIsOpen;

    private boolean fridayIsOpen;

    private boolean saturdayIsOpen;

    private boolean sundayIsOpen;


    public OpeningHoursCreationDTO(String username, LocalTime mondayOpeningHours, LocalTime mondayClosingHours, LocalTime tuesdayOpeningHours, LocalTime tuesdayClosingHours, LocalTime wednesdayOpeningHours, LocalTime wednesdayClosingHours, LocalTime thursdayOpeningHours, LocalTime thursdayClosingHours, LocalTime fridayOpeningHours, LocalTime fridayClosingHours, LocalTime saturdayOpeningHours, LocalTime saturdayClosingHours, LocalTime sundayOpeningHours, LocalTime sundayClosingHours, boolean mondayIsOpen, boolean tuesdayIsOpen, boolean wednesdayIsOpen, boolean thursdayIsOpen, boolean fridayIsOpen,  boolean saturdayIsOpen, boolean sundayIsOpen) {
        this.mondayOpeningHours = mondayOpeningHours;
        this.tuesdayOpeningHours = tuesdayOpeningHours;
        this.wednesdayOpeningHours = wednesdayOpeningHours;
        this.thursdayOpeningHours = thursdayOpeningHours;
        this.fridayOpeningHours = fridayOpeningHours;
        this.saturdayOpeningHours = saturdayOpeningHours;
        this.sundayOpeningHours = sundayOpeningHours;
        this.mondayClosingHours = mondayClosingHours;
        this.tuesdayClosingHours = tuesdayClosingHours;
        this.wednesdayClosingHours = wednesdayClosingHours;
        this.thursdayClosingHours = thursdayClosingHours;
        this.fridayClosingHours = fridayClosingHours;
        this.saturdayClosingHours = saturdayClosingHours;
        this.sundayClosingHours = sundayClosingHours;
        this.username = username;
        this.mondayIsOpen=mondayIsOpen;
        this.tuesdayIsOpen=tuesdayIsOpen;
        this.wednesdayIsOpen=wednesdayIsOpen;
        this.thursdayIsOpen=thursdayIsOpen;
        this.fridayIsOpen=fridayIsOpen;
        this.saturdayIsOpen=saturdayIsOpen;
        this.sundayIsOpen=sundayIsOpen;
    }

    public LocalTime getMondayOpeningHours() {
        return mondayOpeningHours;
    }

    public LocalTime getTuesdayOpeningHours() {
        return tuesdayOpeningHours;
    }

    public LocalTime getWednesdayOpeningHours() {
        return wednesdayOpeningHours;
    }

    public LocalTime getThursdayOpeningHours() {
        return thursdayOpeningHours;
    }

    public LocalTime getFridayOpeningHours() {
        return fridayOpeningHours;
    }

    public LocalTime getSaturdayOpeningHours() {
        return saturdayOpeningHours;
    }

    public LocalTime getSundayOpeningHours() {
        return sundayOpeningHours;
    }

    public LocalTime getMondayClosingHours() {
        return mondayClosingHours;
    }

    public LocalTime getTuesdayClosingHours() {
        return tuesdayClosingHours;
    }

    public LocalTime getWednesdayClosingHours() {
        return wednesdayClosingHours;
    }

    public LocalTime getThursdayClosingHours() {
        return thursdayClosingHours;
    }

    public LocalTime getFridayClosingHours() {
        return fridayClosingHours;
    }

    public LocalTime getSaturdayClosingHours() {
        return saturdayClosingHours;
    }

    public LocalTime getSundayClosingHours() {
        return sundayClosingHours;
    }

    public String getUsername() {
        return username;
    }


    @Override
    public String toString() {
        return "OpeningHoursCreationDTO{" +
                "mondayOpeningHours=" + mondayOpeningHours +
                ", tuesdayOpeningHours=" + tuesdayOpeningHours +
                ", wednesdayOpeningHours=" + wednesdayOpeningHours +
                ", thursdayOpeningHours=" + thursdayOpeningHours +
                ", fridayOpeningHours=" + fridayOpeningHours +
                ", saturdayOpeningHours=" + saturdayOpeningHours +
                ", sundayOpeningHours=" + sundayOpeningHours +
                '}';
    }

    public boolean isMondayIsOpen() {
        return mondayIsOpen;
    }

    public boolean isTuesdayIsOpen() {
        return tuesdayIsOpen;
    }

    public boolean isWednesdayIsOpen() {
        return wednesdayIsOpen;
    }

    public boolean isThursdayIsOpen() {
        return thursdayIsOpen;
    }

    public boolean isFridayIsOpen() {
        return fridayIsOpen;
    }

    public boolean isSaturdayIsOpen() {
        return saturdayIsOpen;
    }

    public boolean isSundayIsOpen() {
        return sundayIsOpen;
    }
}

File: [CODE]/rightoversBlazorNWEB/Domain/DTOs/OpeningHoursCreationDto.cs
using Domain.Classes;

namespace Domain.DTOs;

public class OpeningHoursCreationDto
{
    public OpeningHoursCreationDto(string username, MyTime mondayOpeningHours, MyTime mondayClosingHours,
        MyTime tuesdayOpeningHours,
        MyTime tuesdayClosingHours, MyTime wednesdayOpeningHours, MyTime wednesdayClosingHours,
        MyTime thursdayOpeningHours, MyTime thursdayClosingHours, MyTime fridayOpeningHours, MyTime fridayClosingHours,
        MyTime saturdayOpeningHours, MyTime saturdayClosingHours, MyTime sundayOpeningHours, MyTime sundayClosingHours,
        bool isMondayOpen, bool isTuesdayOpen, bool isWednesdayOpen, bool isThursdayOpen, bool isFridayOpen,
        bool isSaturdayOpen, bool isSundayOpen)
    {
        Username = username;
        MondayOpeningHours = mondayOpeningHours;
        MondayClosingHours = mondayClosingHours;
        TuesdayOpeningHours = tuesdayOpeningHours;
        TuesdayClosingHours = tuesdayClosingHours;
        WednesdayOpeningHours = wednesdayOpeningHours;
        WednesdayClosingHours = wednesdayClosingHours;
        ThursdayOpeningHours = thursdayOpeningHours;
        ThursdayClosingHours = thursdayClosingHours;
        FridayOpeningHours = fridayOpeningHours;
        FridayClosingHours = fridayClosingHours;
        SaturdayOpeningHours = saturdayOpeningHours;
        SaturdayClosingHours = saturdayClosingHours;
        SundayOpeningHours = sundayOpeningHours;
        SundayClosingHours = sundayClosingHours;
        IsMondayOpen = isMondayOpen;
        IsTuesdayOpen = isTuesdayOpen;
        IsWednesdayOpen = isWednesdayOpen;
        IsThursdayOpen = isThursdayOpen;
        IsFridayOpen = isFridayOpen;
        IsSaturdayOpen = isSaturdayOpen;
        IsSundayOpen = isSundayOpen;
    }

    public OpeningHoursCreationDto()
    {
    }

    public MyTime MondayOpeningHours { get; set; }
    public MyTime MondayClosingHours { get; set; }
    public MyTime TuesdayOpeningHours { get; set; }
    public MyTime TuesdayClosingHours { get; set; }
    public MyTime WednesdayOpeningHours { get; set; }
    public MyTime WednesdayClosingHours { get; set; }
    public MyTime ThursdayOpeningHours { get; set; }
    public MyTime ThursdayClosingHours { get; set; }
    public MyTime FridayOpeningHours { get; set; }
    public MyTime FridayClosingHours { get; set; }
    public MyTime SaturdayOpeningHours { get; set; }
    public MyTime SaturdayClosingHours { get; set; }
    public MyTime SundayOpeningHours { get; set; }
    public MyTime SundayClosingHours { get; set; }

    public bool IsMondayOpen { get; set; }

    public bool IsTuesdayOpen { get; set; }

    public bool IsWednesdayOpen { get; set; }

    public bool IsThursdayOpen { get; set; }

    public bool IsFridayOpen { get; set; }

    public bool IsSaturdayOpen { get; set; }

    public bool IsSundayOpen { get; set; }
    public string Username { get; set; }
}